src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/TextEncoder.cs

```diff
@@ -132,9 +139,18 @@ public virtual string Encode(string value)
         // NOTE: The order of the parameters to this method is a work around for https://github.com/dotnet/corefx/issues/4455
         // and the underlying Mono bug: https://bugzilla.xamarin.com/show_bug.cgi?id=36052.
         // If changing the signature of this method, ensure this issue isn't regressing on Mono.
-        private unsafe int EncodeIntoBuffer(char* buffer, int bufferLength, char* value, int valueLength, int firstCharacterToEncode)
+        private unsafe OperationStatus EncodeIntoBuffer(
+            char* buffer,
+            int bufferLength,
+            char* value,
+            int valueLength,
+            out int charsConsumed,
+            out int charsWritten,
+            int firstCharacterToEncode,
+            bool isFinalBlock = true)
         {
-            int totalWritten = 0;
+            char* originalBuffer = buffer;
+            charsWritten = 0;
```

Comment:
Nit: You'll get a slight perf boost if you create a temporary local for this value and don't assign the _out_ parameter until the very end of the method. This could allow the local to be eligible for enregistration.

Reply:
Sure. Thanks

Reply:
Update: I will keep the code as-is. On my local build (Windows + release) it is faster to write to the `out` parameter. I verified with a benchmark.

The three cases with a local:
```cs
charsWrittenLocal += firstCharacterToEncode;
00007FF835CF6299  mov         esi,r15d  
---
charsWrittenLocal++;
00007FF83BAE4A97  mov         r9d,dword ptr [rsp+6Ch]  
00007FF83BAE4A9C  inc         r9d  
00007FF83BAE4A9F  mov         dword ptr [rsp+6Ch],r9d  
---
                    charsWrittenLocal += charsWrittenThisTime;
00007FF83BAE4B39  mov         r10d,dword ptr [rsp+6Ch]  
00007FF83BAE4B3E  add         r10d,dword ptr [rsp+50h]  
00007FF83BAE4B43  mov         dword ptr [rsp+6Ch],r10d  
```

With the `out` parameter:
```cs
                charsWritten += firstCharacterToEncode;
00007FF835CF47FC  add         dword ptr [r14],r12d  
---
 charsWritten++;
00007FF835AD7AD4  inc         dword ptr [r14]  
---
                    charsWritten += charsWrittenThisTime;
00007FF835AD7B6E  mov         eax,dword ptr [rsp+48h]  
00007FF835AD7B72  add         dword ptr [r14],eax  
```